pipelines/broad/arrays/imputation/Imputation_to_TDR.wdl

```diff
@@ -0,0 +1,254 @@
+        imputed_multisample_vcf\timputed_multisample_vcf_index\t\
+        imputed_single_sample_vcfs\timputed_single_sample_vcf_indices\t\
+        chip_well_barcodes" \
+        > ingestDataset_imputation_outputs.tsv
+
+        # handle array[type] variables to print as list with double quotes
+        imputed_single_sample_vcfs='~{sep='","' imputed_single_sample_vcfs}'
+        echo "imputed_single_sample_vcfs"
+        echo "[\"${imputed_single_sample_vcfs}\"]"
+
+        imputed_single_sample_vcf_indices='~{sep='","' imputed_single_sample_vcf_indices}'
+        echo "imputed_single_sample_vcf_indices"
+        echo "[\"${imputed_single_sample_vcf_indices}\"]"
+
+        chip_well_barcodes='~{sep='","' chip_well_barcodes}'
+        echo "chip_well_barcodes"
+        echo "[\"${chip_well_barcodes}\"]"
+
+        # write file paths to row in tsv file
+        echo -e "~{aggregated_imputation_metrics}\t~{chunks_info}\t~{failed_chunks}\t~{n_failed_chunks}\t\
+        ~{imputed_multisample_vcf}\t~{imputed_multisample_vcf_index}\t\
+        [\"${imputed_single_sample_vcfs}\"]\t\
+        [\"${imputed_single_sample_vcf_indices}\"]\t\
+        [\"${chip_well_barcodes}\"]" \
+        >> ingestDataset_imputation_outputs.tsv
+
+
+        python3 << CODE
+        import pandas as pd
+
+        tsv_df = pd.read_csv("ingestDataset_imputation_outputs.tsv", sep="\t")
+        tsv_df = tsv_df.dropna(axis=1, how="all")  # drop columns if no value (optional outputs etc)
+
+        outputs = tsv_df.to_json("ingestDataset_imputation_outputs.json", orient="records")  # write json file
+
+        CODE
+    >>>
+
+    runtime {
+        docker: "broadinstitute/horsefish:emerge_scripts"
+    }
+
+    output {
+        File ingest_outputs_tsv = "ingestDataset_imputation_outputs.tsv"
+    }
+}
+
+task format_imputation_wide_outputs{
+    input {
+        Array[String]?  imputed_single_sample_vcfs
+        Array[String]?  imputed_single_sample_vcf_indices
+    }
+
+    String prefix="[\""
+    String postfix="\"]"
+
+    command <<<
+
+        python3 << CODE
+        import pandas as pd
+        import os
+
+        print("imputed_single_sample_vcfs")
+        single_sample_vcfs=~{prefix}~{sep="\", \"" imputed_single_sample_vcfs}~{postfix}
+        print(type(single_sample_vcfs))
+        print(single_sample_vcfs)
+
+        print("imputed_single_sample_vcf_indices")
+        single_sample_vcf_indices=~{prefix}~{sep="\", \"" imputed_single_sample_vcf_indices}~{postfix}
+        print(type(single_sample_vcf_indices))
+        print(single_sample_vcf_indices)
+
+        print("creating dataframe")
+        all_samples = []
+        sample_dict = {}
+
+        print("getting vcf + vcf index file names and paths and chipwell barcode")
+        # for each file in list of imputed vcfs, get chip_well_barcode value
+        for vcf in single_sample_vcfs:
+            imputed_vcf_filename = vcf.split("/")[-1]
```

Comment:
I think extracting the chip_well_barcode from the filename is fine for now, but probably want to change to getting it from the sample id at some point.  Or something.  We really need to come up with a more robust solution for how these keys get passed around, but not in the scope of this PR.

Reply:
the sample_id value that is getting passed into the pipeline (from TDR) does not have any indication of chipwell - are you thinking about using sample_id and maybe calling mercury from within the WDL to grab the chipwell?